packages/ldc/build.sh

```diff
@@ -0,0 +1,122 @@
+TERMUX_PKG_EXTRA_CONFIGURE_ARGS="
+-DLLVM_ENABLE_PIC=ON
+-DLLVM_BUILD_TOOLS=OFF
+-DLLVM_BUILD_UTILS=OFF
+-DPYTHON_EXECUTABLE=`which python`
+-DLLVM_TABLEGEN=$TERMUX_PKG_HOSTBUILD_DIR/bin/llvm-tblgen"
+TERMUX_PKG_KEEP_STATIC_LIBRARIES=true
+TERMUX_PKG_NO_DEVELSPLIT=yes
+TERMUX_PKG_MAINTAINER="Joakim @joakim-noah"
+
+termux_step_post_extract_package () {
+	local LLVM_SRC_VERSION=3.9.1
+	termux_download \
+		http://llvm.org/releases/${LLVM_SRC_VERSION}/llvm-${LLVM_SRC_VERSION}.src.tar.xz \
+		$TERMUX_PKG_CACHEDIR/llvm-${LLVM_SRC_VERSION}.src.tar.xz \
+		1fd90354b9cf19232e8f168faf2220e79be555df3aa743242700879e8fd329ee
+
+	tar xf $TERMUX_PKG_CACHEDIR/llvm-${LLVM_SRC_VERSION}.src.tar.xz
+	mv llvm-${LLVM_SRC_VERSION}.src llvm
+
+	BOOTSTRAP_COMPILER_VERSION=1.3.0-beta1
+	termux_download \
+		https://github.com/joakim-noah/android/releases/download/tea/ldc2-android-arm-${BOOTSTRAP_COMPILER_VERSION}-linux-x86_64.tar.xz \
+		$TERMUX_PKG_CACHEDIR/ldc2-android-arm-${BOOTSTRAP_COMPILER_VERSION}-linux-x86_64.tar.xz \
+		5ddd141418f5c686a82aaa925a82c21e9de747da1a3a6f30cab00a1c80b15c6e
+
+	sed "s#\@TERMUX_PKG_HOSTBUILD\@#$TERMUX_PKG_HOSTBUILD_DIR#" $TERMUX_PKG_BUILDER_DIR/ldc-linker-flags.patch.in > \
+		$TERMUX_PKG_BUILDER_DIR/ldc-linker-flags.patch
+	sed "s#\@TERMUX_PKG_BUILD\@#$TERMUX_PKG_BUILDDIR#" $TERMUX_PKG_BUILDER_DIR/ldc-llvm-config.patch.in > \
+		$TERMUX_PKG_BUILDER_DIR/ldc-llvm-config.patch
+	sed -i "s#\@TERMUX_PKG_SRC\@#$TERMUX_PKG_SRCDIR#" $TERMUX_PKG_BUILDER_DIR/ldc-llvm-config.patch
+}
+
+termux_step_host_build () {
+	tar xf $TERMUX_PKG_CACHEDIR/ldc2-android-arm-${BOOTSTRAP_COMPILER_VERSION}-linux-x86_64.tar.xz -C $TERMUX_PKG_HOSTBUILD_DIR
+	mv $TERMUX_PKG_HOSTBUILD_DIR/ldc2-android-arm-${BOOTSTRAP_COMPILER_VERSION}-linux-x86_64 ldc-bootstrap
+
+	termux_setup_cmake
+	cmake -G "Unix Makefiles" $TERMUX_PKG_SRCDIR/llvm
+	make -j $TERMUX_MAKE_PROCESSES llvm-tblgen
+}
+
+termux_step_pre_configure () {
+	rm $TERMUX_PKG_BUILDER_DIR/ldc-linker-flags.patch
+	rm $TERMUX_PKG_BUILDER_DIR/ldc-llvm-config.patch
+
+	local LLVM_DEFAULT_TARGET_TRIPLE=$TERMUX_HOST_PLATFORM
+	if [ $TERMUX_ARCH = "arm" ]; then
+		LLVM_TARGET_ARCH=ARM
+		# See https://github.com/termux/termux-packages/issues/282
+		LLVM_DEFAULT_TARGET_TRIPLE="armv7a-linux-androideabi"
+	else
+		termux_error_exit "Invalid arch: $TERMUX_ARCH"
+	fi
+        # see CMakeLists.txt and tools/clang/CMakeLists.txt
+	TERMUX_PKG_EXTRA_CONFIGURE_ARGS+=" -DLLVM_DEFAULT_TARGET_TRIPLE=$LLVM_DEFAULT_TARGET_TRIPLE"
+	TERMUX_PKG_EXTRA_CONFIGURE_ARGS+=" -DLLVM_TARGET_ARCH=$LLVM_TARGET_ARCH	-DLLVM_TARGETS_TO_BUILD=$LLVM_TARGET_ARCH"
+
+	# CPPFLAGS adds the system llvm to the include path, which causes
+	# conflicts with the local patched llvm when compiling ldc
+	CPPFLAGS=""
+
+	OLD_TERMUX_PKG_SRCDIR=$TERMUX_PKG_SRCDIR
+	TERMUX_PKG_SRCDIR=$TERMUX_PKG_SRCDIR/llvm
+
+	OLD_TERMUX_PKG_BUILDDIR=$TERMUX_PKG_BUILDDIR
+	TERMUX_PKG_BUILDDIR=$TERMUX_PKG_BUILDDIR/llvm
+	mkdir "$TERMUX_PKG_BUILDDIR"
```

Comment:
This prelude is necessary because I use `termux_step_configure` to configure the locally patched llvm 3.9.1, then configure ldc in the `post_configure`.